DownloadAllSongsAndAlbums.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+mkdir -p $HOME/Music/SongLibrary
+mkdir -p $HOME/Pictures/Covers
+cd $HOME/Music/SongLibrary
+
+export AWS_DEFAULT_REGION=us-west-2
+export AWS_DEFAULT_OUTPUT=text
+
+readarray -t song <<< "$(aws s3 ls s3://ucsolarteam.onboardmedia.songs | awk '{for (i=4; i<NF; i++) printf $i " "; print $NF}' | sed s/\ /\+/g | rev | cut -c 5- | rev )"
+albumlist=()
+
+for i in "${song[@]}"; do
+	wget https://s3-us-west-2.amazonaws.com/ucsolarteam.onboardmedia.songs/$i.mp3
+done 
+
+album="$(eyeD3 ${song[0]}.mp3 | grep -o -P '(?<=: ).*(?=year)')"
+album=${album//[[:blank:]]/}
+album=${album::-4}
+albumlist+=($album)
+uniquealbumlist+=($album)
+album="$HOME/Pictures/Covers/$album.jpg"
+eyeD3 --write-images=$PWD ${song[0]}.mp3
+mv OTHER.JPG $album
+
+
+i=1
+while [ $i -lt ${#song[@]} ]; do 
```

Comment:
Correct me if I'm wrong, but it looks like this `while` block is to save unique images? 

You can put functions inside bash scripts, I think this block would be nice to have, to better clarify what's going on. The function should have a name to better show that it deals with the images.